litestar/utils/typing.py

```diff
@@ -163,14 +172,30 @@ def unwrap_annotation(annotation: Any) -> tuple[Any, tuple[Any, ...], set[Any]]:
     Returns:
         A tuple of the unwrapped annotation and any ``Annotated`` metadata, and a set of any wrapper types encountered.
     """
-    origin = get_origin(annotation)
-    wrappers = set()
+
     metadata = []
-    while origin in wrapper_type_set:
-        wrappers.add(origin)
-        annotation, *meta = get_args(annotation)
-        metadata.extend(meta)
-        origin = get_origin(annotation)
+    wrappers = set()
+
+    stack = [annotation]
+
+    while stack:
+        ann = stack.pop()
+
+        if isinstance(ann, TypeAliasType):
+            wrappers.add(TypeAliasType)
+            stack.append(ann.__value__)
```

Comment:
This line can raise all kinds of exceptions. Do we have cases like `NameError` and `TypeError` covered in tests?

Reply:
Could you elaborate what kind of exception scenarios you're thinking of here? My thinking was that, if this does raise an exception, it's okay to let it bubble up since we can't properly deal with it

Reply:
`type A = Undefined`

Yeah, I agree that it is fine, we just need to test these cases.